frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -2344,6 +2347,29 @@ func validateBaseImagePlatform(name string, expected, actual ocispecs.Platform,
 	}
 }
 
+func validateNoSecretKey(key string, location []parser.Range, lint *linter.Linter) {
+	// Check for either full value or first/last word.
+	// Examples: api_key, DATABASE_PASSWORD, GITHUB_TOKEN, secret_MESSAGE, AUTH
+	// Case insensitive.
+	secretTokens := []string{
+		"apikey",
+		"auth",
+		"credential",
+		"credentials",
+		"key",
+		"password",
+		"pword",
+		"passwd",
+		"secret",
+		"token",
+	}
+	pattern := `(?i)(?:_|^)(?:` + strings.Join(secretTokens, "|") + `)(?:_|$)`
+	if matched, _ := regexp.MatchString(pattern, key); matched {
```

Comment:
This needs to be initialized with `sync.Once+regexp.MustCompile` and then reused. Initial conversion from string is much more expensive than the matching.